File: problems/js/buySellStock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let minSoFar = Number.MAX_VALUE;
  let maxProf = 0;

  for (var stock of prices) {
      if (stock < minSoFar) {
          minSoFar = stock;
      }

      if (stock - minSoFar > maxProf) {
          maxProf = stock - minSoFar;
      }
  }
  return maxProf;
};

File: problems/python/buySellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        # Init values
        maxProf = 0
        minSoFar = math.inf

        for p in prices:

            minSoFar = min(minSoFar, p)
            maxProf = max(maxProf, p - minSoFar)

        return maxProf